src/main/java/nl/workingtalent/bookrental/controller/BookController.java

```diff
@@ -69,8 +84,38 @@ public Map<String, String> delete(@RequestHeader(name = "Authorization") String
 			throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid permissions for deleting book");
 		}
 
-		bookRepo.deleteById(id);
+		Book book = bookRepo.findById(id).get();
+
+		for (Copy copy : book.getCopies()) {
+
+			// Disable copy
+			copy.setInService(false);
+			copyRepo.save(copy);
+
+			// End active loan
+			for (Loan loan : copy.getLoans()) {
+
+				// Copy is currently being rented and does NOT have an end date
+				if (loan.getStartDate() != null && !loan.getStartDate().equalsIgnoreCase("")) {
+					if (loan.getEndDate() == null || loan.getEndDate().equalsIgnoreCase("")) {
```

Comment:
Kan dit niet in 1 if statement?

Reply:
In theorie ja, alleen ze doen wel echt iets anders. De 1e checkt alleen of er een startdatum is. Zodra deze er is, is er dus een actieve loan. Dan kijkt de 2e if statement of er ook een einddatum is. Het zou in 1 if statement kunnen in dezelfde volgorde, maar dan zijn er 4 && cases in 1 regel, wordt vrij onoverzichtelijk ;)

Reply:
Het is overigens een prima gewoontje om if statements precies andersom te doen. In de context van bijvoorbeeld inloggen:

if (verkeerde wachtwoord) return Error;
if (user bestaat niet) return Error2;
if (connection timeout) return Error3;

// Voer daadwerkelijk login uit na alle checks
logIn();